### PR TITLE
feat: solve `fixed_base_scalar_mul` black box functions in rust

### PR DESCRIPTION
**Pull request by TomAFrench**
# Description

## Problem\*

Resolves 

## Summary\*

This PR adds a rust-based solver for `fixed_base_scalar_mul` black box functions so we don't need to rely on the WASM for this.

This does impose a restriction that the curve being used is bn254 but this seems like something we can accept in the short term. If someone was using bls12_381 and wanted to perform fixed base scalar multiplication then we can add in a solver implementation for that curve.

## Documentation

- [ ] This PR requires documentation updates when merged.

  

  - [ ] I will submit a noir-lang/docs PR.

  

  - [ ] I will request for and support Dev Rel's help in documenting this PR.

  
  

## Additional Context

# PR Checklist\*

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: acvm-repo/blackbox_solver/Cargo.toml [hunk 1]
```diff
@@ -33,6 +34,8 @@ p256 = { version = "0.11.0", features = [
     "digest",
     "arithmetic",
 ] }
+grumpkin = { git = "https://github.com/noir-lang/grumpkin", features = ["std"] }
```
Comment (kevaundray) [inline acvm-repo/blackbox_solver/Cargo.toml:37]: We should open up an issue to use the arkworks version; back when this was made they did not have grumpkin, they recently added it a month ago

Comment (kevaundray) [reply acvm-repo/blackbox_solver/Cargo.toml:37]: Not sure if all of the parameters match, so might be better to make things work with our version which we can modify if it doesn't match

Comment (TomAFrench) [reply acvm-repo/blackbox_solver/Cargo.toml]: Yeah, we can't do this at the minute. arkworks haven't pushed this crate to crates.io and using a git dependency results in issues due to it wanting a different version of `ark-bn254`

**Comment by TomAFrench** [comment]
I'm considering whether we want to keep the `BlackBoxFunctionSolver` trait going forwards even after we have rust implementation of all of the black box functions. `barretenberg_blackbox_solver` would then transition into `bn254_blackbox_solver` and alternative curves can provide their own implementations of the opcodes which embed knowledge about the underlying curve.

**Comment by kevaundray** [comment]
> I'm considering whether we want to keep the `BlackBoxFunctionSolver` trait going forwards even after we have rust implementation of all of the black box functions. `barretenberg_blackbox_solver` would then transition into `bn254_blackbox_solver` and alternative curves can provide their own implementations of the opcodes which embed knowledge about the underlying curve.

Sounds like a good idea

**Review by kevaundray — APPROVED** [review]

**Pull request merged**